source/d3d12/d3d12_device.hpp

```diff
@@ -95,9 +98,24 @@ struct __declspec(uuid("2523AFF4-978B-4939-BA16-8EE876A4CB2A")) D3D12Device : ID
 	D3D12_DRIVER_MATCHING_IDENTIFIER_STATUS STDMETHODCALLTYPE CheckDriverMatchingIdentifier(D3D12_SERIALIZED_DATA_TYPE SerializedDataType, const D3D12_SERIALIZED_DATA_DRIVER_MATCHING_IDENTIFIER *pIdentifierToCheck) override;
 	#pragma endregion
 
+	void add_commandlist_trackers(ID3D12CommandList* command_list, reshade::d3d12::draw_call_tracker &tracker_source);
+	void merge_commandlist_trackers(ID3D12CommandList* command_list, reshade::d3d12::draw_call_tracker &tracker_destination);
+
 	bool check_and_upgrade_interface(REFIID riid);
+	void clear_drawcall_stats(bool all = false);
+	void track_active_rendertargets(ID3D12GraphicsCommandList *pcmdList, D3D12_CPU_DESCRIPTOR_HANDLE pDepthStencilView);
 
+#if RESHADE_DX12_CAPTURE_DEPTH_BUFFERS
+	bool save_depth_texture(ID3D12GraphicsCommandList *pcmdList, D3D12_CPU_DESCRIPTOR_HANDLE pDepthStencilView, bool cleared);
+	void track_cleared_depthstencil(ID3D12GraphicsCommandList *pcmdList, D3D12_CLEAR_FLAGS ClearFlags, D3D12_CPU_DESCRIPTOR_HANDLE pDepthStencilView);
+#endif
+	
 	LONG _ref = 1;
 	ID3D12Device *_orig;
 	unsigned int _interface_version;
+	std::vector<std::shared_ptr<reshade::d3d12::runtime_d3d12>> _runtimes;
+	std::unordered_map<ID3D12CommandList *, reshade::d3d12::draw_call_tracker> _trackers_per_commandlist;
+	std::mutex _trackers_per_commandlist_mutex;
+	reshade::d3d12::draw_call_tracker _draw_call_tracker;
+	std::mutex _clear_DSV_iter_mutex;
```

Comment:
This is not necessary if `clear_DSV_iter` is an atomic.

Reply:
Changed